grails-app/services/au/org/ala/ecodata/DocumentService.groovy

```diff
@@ -360,4 +375,13 @@ class DocumentService {
 
         results.collect{toMap(it, 'flat')}
     }
+
+    Boolean isMobileAppForProject(Map project){
+        List links = project.links
+        Boolean isMobileApp = false;
+        links?.each { link ->
+            isMobileApp = link.role in MOBILE_APP_ROLE;
```

Comment:
This will only return the correct result if the last link is a mobile app. Each iteration is overwriting the ```isMobelApp``` value.
e.g. [mobile, notMobile] will return false;
[notMobile, mobile] will return true.

If you want to know if there are ANY mobile apps, then change it to ``` boolean isMobileApp = links?.any { it.role in MOBILE_APP_ROLE }``` 

If you want to know if ALL the links are mobile, then change it to ```boolean isMobileApp = links?.every { it.role in MOBILE_APP_ROLE }```